packages/twenty-server/src/database/commands/upgrade-version/0-40/0-40-phone-calling-code.command.ts

```diff
@@ -0,0 +1,344 @@
+import { WorkspaceMigrationFactory } from 'src/engine/metadata-modules/workspace-migration/workspace-migration.factory';
+import { WorkspaceMigrationService } from 'src/engine/metadata-modules/workspace-migration/workspace-migration.service';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import { computeObjectTargetTable } from 'src/engine/utils/compute-object-target-table.util';
+import { WorkspaceMigrationRunnerService } from 'src/engine/workspace-manager/workspace-migration-runner/workspace-migration-runner.service';
+import { isDefined } from 'src/utils/is-defined';
+
+const callingCodeToCountryCode = (callingCode: string): string => {
+  if (!callingCode) return '';
+  let callingCodeWithoutPlus = callingCode;
+
+  if (callingCode.startsWith('+')) {
+    callingCodeWithoutPlus = callingCode.slice(1);
+  }
+
+  return (
+    getCountries().find(
+      (countryCode) =>
+        getCountryCallingCode(countryCode) === callingCodeWithoutPlus,
+    ) || ''
+  );
+};
+
+// console.log('33', callingCodeToCountryCode('33'));
+// console.log('+33', callingCodeToCountryCode('+33'));
+// console.log('', callingCodeToCountryCode(''));
+// console.log('FR', callingCodeToCountryCode('FR'));
+
+const isCallingCode = (callingCode: string): boolean => {
+  return callingCodeToCountryCode(callingCode) !== '';
+};
+
+// console.log('33', isCallingCode('33'));
+// console.log('+33', isCallingCode('+33'));
+// console.log('', isCallingCode(''));
+// console.log('FR', isCallingCode('FR'));
+
+@Command({
+  name: 'upgrade-0.40:phone-calling-code',
+  description: 'Add calling code and change country code with default one',
+})
+export class PhoneCallingCodeCommand extends ActiveWorkspacesCommandRunner {
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    protected readonly workspaceRepository: Repository<Workspace>,
+    @InjectRepository(FieldMetadataEntity, 'metadata')
+    private readonly fieldMetadataRepository: Repository<FieldMetadataEntity>,
+    @InjectRepository(ObjectMetadataEntity, 'metadata')
+    private readonly objectMetadataRepository: Repository<ObjectMetadataEntity>,
+    private readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    private readonly workspaceMigrationService: WorkspaceMigrationService,
+    private readonly workspaceMigrationFactory: WorkspaceMigrationFactory,
+    private readonly workspaceMigrationRunnerService: WorkspaceMigrationRunnerService,
+    private readonly workspaceMetadataVersionService: WorkspaceMetadataVersionService,
+  ) {
+    super(workspaceRepository);
+  }
+
+  async executeActiveWorkspacesCommand(
+    _passedParam: string[],
+    _options: ActiveWorkspacesCommandOptions,
+    workspaceIds: string[],
+  ): Promise<void> {
+    this.logger.log(
+      'Running command to add calling code and change country code with default one',
+    );
+
+    // Prerequisite : the field callingcode was creted in the codebase (PHONES type modification)
```

Comment:
I don't think we need to keep those comments, however I guess you can simplify a bit by splitting your code into smaller functions 🙂

Reply:
Since it's a command, I would not spend too much time refactoring into smaller function if that's ok with you.
About the comments, I will delete them. I usually like having an idea of what we are about to achieve, but I understand that there is a "no comment" rule on twenty